eclair-core/src/main/scala/fr/acinq/eclair/db/jdbc/JdbcUtils.scala

```diff
@@ -60,6 +60,42 @@ trait JdbcUtils {
     }
   }
 
+  private def createVersionTable(statement: Statement): Unit = {
+    statement.executeUpdate("CREATE TABLE IF NOT EXISTS versions (db_name TEXT NOT NULL PRIMARY KEY, version INTEGER NOT NULL)")
+  }
+
+  /**
+   * Several logical databases (channels, network, peers) may be stored in the same physical sqlite database.
+   * We keep track of their respective version using a dedicated table. The version entry will be created if
+   * there is none but will never be updated here (use setVersion to do that).
+   */
+  def getVersion(statement: Statement, db_name: String): Option[Int] = {
+    createVersionTable(statement)
+    // if there was a previous version installed, this will return a different value from current version
+    val rs = statement.executeQuery(s"SELECT version FROM versions WHERE db_name='$db_name'")
+    if (rs.next()) Some(rs.getInt("version")) else None
+  }
+
+  /**
+   * Updates the version for a particular logical database, it will overwrite the previous version.
+   *
+   * NB: we could define this method in [[fr.acinq.eclair.db.sqlite.SqliteUtils]] and [[fr.acinq.eclair.db.pg.PgUtils]]
+   *     but it would make testing more complicated because we need to use one or the other depending on the backend.
```

Comment:
This design choice is debatable but I think it's worthwhile, less duplication and simpler tests.

Reply:
Since this is a shared mechanism for all DB types, I think it's perfectly fine to have it here, especially if it makes testing easier.